### PR TITLE
WDIP-16: enable hotkeys for selecting flyaway menu options

### PR DESCRIPTION
**Pull request by utiq**
I introduced [react-use](https://github.com/streamich/react-use), which is a collection of essential hooks, I use it in every app I develop. I saw there are multiple parts of the webDip app that it would be nice to reuse.

### CODE REVIEW

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx [hunk 1]
```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
...
-        clickHandler={clickHandler("Hold")}
+        clickHandler={() => clickHandler("Hold")}
       />
       <WDFlyoutButton
         province={province}
         unitSlotName={unitSlotName}
         position="right"
         text="Move"
-        clickHandler={clickHandler("Move")}
+        clickHandler={() => clickHandler("Move")}
```
Comment (adamlerer) [inline beta-src/src/components/map/components/WDFlyoutContainer.tsx:100]: Good catch - how was this working before?

**Review by adamlerer — APPROVED** [review]

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx [hunk 2]
```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
...
+    if (!order.inProgress) return;
+
+    if (fleetAndSeaCondition()) {
+      clickHandler("Convoy");
+    }
+  };
+
+  const armyCondition = (): boolean =>
+    unit?.unit?.type === "Army" &&
+    legalOrders.legalViasByUnitID[order.unitID].length > 0;
```
Comment (adamlerer) [inline beta-src/src/components/map/components/WDFlyoutContainer.tsx:60]:
```suggestion
  const canVia = 
    unit?.unit?.type === "Army" &&
    legalOrders.legalViasByUnitID[order.unitID].length > 0;
```

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx [hunk 3]
```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
...
+
+  const moveAction = () => {
+    if (!order.inProgress) return;
+
+    if (armyCondition()) {
+      clickHandler("Move", "Yes");
+    } else {
+      clickHandler("Move");
+    }
+  };
```
[the remark this answers is not in the document]
Comment (adamlerer) [reply beta-src/src/components/map/components/WDFlyoutContainer.tsx:70]: Maybe we need to have more discussion, but either via's should have the hotkey "v" only, or you could use "c" (becauseyou can never do both a convoy and a via). But you can't overlap it with a regular move.

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx [hunk 4]
```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
...
+        type: orderType,
+        viaConvoy,
+      }),
+    );
+  };
+
+  const fleetAndSeaCondition = (): boolean =>
+    unit?.unit?.type === "Fleet" &&
+    mTerr.provinceMapData.type === "Sea" &&
+    legalOrders.hasAnyLegalConvoysByUnitID[order.unitID];
```
[the remark this answers is not in the document]
Comment (utiq) [reply beta-src/src/components/map/components/WDFlyoutContainer.tsx:48]: updated

File: beta-src/src/components/map/components/WDFlyoutContainer.tsx (hunk 2 above)
Comment (utiq) [reply beta-src/src/components/map/components/WDFlyoutContainer.tsx:60]: updated

**Pull request merged**